### PR TITLE
Update mqtt destination to use `outputRecordCollector` to properly store state

### PR DESCRIPTION
**Pull request by benmoriceau**
## What
Fix MQTT and add test.

### CODE REVIEW

File: airbyte-integrations/connectors/destination-mqtt/src/test/java/io/airbyte/integrations/destination/mqtt/MqttRecordConsumerTest.java [hunk 1]
```diff
@@ -71,18 +100,7 @@ public void testBuildTopicMap(final ConfiguredAirbyteCatalog catalog,
   @Test
   @SuppressWarnings("unchecked")
   void testCannotConnectToBrokers() throws Exception {
-    final MqttDestinationConfig config = MqttDestinationConfig
```
Comment (benmoriceau) [inline airbyte-integrations/connectors/destination-mqtt/src/test/java/io/airbyte/integrations/destination/mqtt/MqttRecordConsumerTest.java:74]: The init is move to the before each method

File: airbyte-integrations/bases/standard-destination-test/build.gradle [hunk 2]
```diff
@@ -11,4 +11,6 @@ dependencies {
     implementation(enforcedPlatform('org.junit:junit-bom:5.8.2'))
     implementation 'org.junit.jupiter:junit-jupiter-api'
     implementation 'org.junit.jupiter:junit-jupiter-params'
+    implementation 'org.junit.jupiter:junit-jupiter-params'
```
Comment (evantahler) [inline airbyte-integrations/bases/standard-destination-test/build.gradle:14]: duplicate of the line above?

File: airbyte-integrations/bases/standard-destination-test/src/main/java/io/airbyte/integrations/standardtest/destination/PerStreamStateMessageTest.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import org.mockito.InOrder;
+import org.mockito.Mockito;
+
+public abstract class PerStreamStateMessageTest {
+  protected abstract Consumer<AirbyteMessage> getMockedConsumer();
+
+  protected abstract FailureTrackingAirbyteMessageConsumer getMessageConsumer();
+
+  @Test
+  void runATest() throws Exception {
```
Comment (evantahler) [inline airbyte-integrations/bases/standard-destination-test/src/main/java/io/airbyte/integrations/standardtest/destination/PerStreamStateMessageTest.java:22]: Nice! this is a super clear test

File: airbyte-integrations/connectors/destination-mqtt/src/main/java/io/airbyte/integrations/destination/mqtt/MqttRecordConsumer.java [hunk 4]
```diff
@@ -75,7 +71,7 @@ protected void startTracked() {
   @Override
   protected void acceptTracked(final AirbyteMessage airbyteMessage) {
     if (airbyteMessage.getType() == AirbyteMessage.Type.STATE) {
-      lastStateMessage = airbyteMessage;
+      outputRecordCollector.accept(airbyteMessage);
```
Comment (evantahler) [inline airbyte-integrations/connectors/destination-mqtt/src/main/java/io/airbyte/integrations/destination/mqtt/MqttRecordConsumer.java:74]: Calling out for future viewers of this PR - this is the main change

**Review by evantahler — APPROVED** [review]
👍

File: airbyte-integrations/bases/standard-destination-test/src/main/java/io/airbyte/integrations/standardtest/destination/PerStreamStateMessageTest.java (hunk 3 above)
Comment (benmoriceau) [reply airbyte-integrations/bases/standard-destination-test/src/main/java/io/airbyte/integrations/standardtest/destination/PerStreamStateMessageTest.java:22]: oh I just realized that I forgot to rename it...

**Comment by benmoriceau** [comment]
Publishing even if test are failing since it is an `alpha` connector.

**Comment by benmoriceau** [comment]
> 

It failed because broken pipe, this looks like a transient error...

File: airbyte-integrations/bases/standard-destination-test/build.gradle (hunk 2 above)
Comment (benmoriceau) [reply airbyte-integrations/bases/standard-destination-test/build.gradle]: updated

**Review by tuliren — APPROVED** [review]
Hmm, no idea why the test failed with broken pipe.

**Pull request closed without merging**